### PR TITLE
doc(applications): add pre-compile/post-compile hook docs

### PR DESCRIPTION
**Pull request by bacongobbler**
refs https://github.com/deis/slugbuilder/pull/86

### CODE REVIEW

File: src/applications/using-buildpacks.md [hunk 1]
```diff
@@ -99,6 +99,21 @@ To use a custom buildpack, set the `BUILDPACK_URL` environment variable.
 
 On your next `git push`, the custom buildpack will be used.
 
+
+## Compile Hooks
+
+Sometimes, an application needs a way to stop or check if a service is running before building an
+app, and sometimes it requires notifying a service that the [Builder][] has finished compiling the
```
Comment (mboersma) [inline src/applications/using-buildpacks.md]: This repeats "sometimes" twice. Maybe rephrase it like:
> Sometimes, an application needs a way to stop or check if a service is running before building an app, which may require notifying a service that...

**Comment by bacongobbler** [comment]
Sorry, missed that comment @mboersma. I'll follow up. :)
